packages/iTwinUI-react/src/core/Table/hooks/useScrollToRow.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import React from 'react';
+import type { Row } from 'react-table';
+import { TableProps } from '../Table';
+
+type ScrollToRow<T extends Record<string, unknown>> = {
+  scrollToIndex: number | undefined;
+  tableRowRef: (row: Row<T>) => (element: HTMLDivElement) => void;
+};
+
+type ScrollToRowProps<T extends Record<string, unknown>> = TableProps<T> & {
+  page: Row<T>[];
+};
+
+export function useScrollToRow<T extends Record<string, unknown>>({
+  enableVirtualization,
+  page,
+  paginatorRenderer,
+  scrollToRow,
+}: ScrollToRowProps<T>): ScrollToRow<T> {
+  const rowRefs = React.useRef<Record<string, HTMLDivElement>>({});
+
+  // For virtualized tables, all we need to do is pass the index of the item
+  // to the VirtualScroll component
+  const scrollToIndex = React.useMemo((): number | undefined => {
+    if (!scrollToRow || paginatorRenderer) {
+      return undefined;
+    }
+
+    const index = scrollToRow(page);
+    return index < 0 ? undefined : index;
+  }, [page, paginatorRenderer, scrollToRow]);
+
+  // For non-virtualized tables, we need to add a ref to each row
+  // and scroll to the element
+  React.useEffect(() => {
+    if (
+      enableVirtualization ||
+      scrollToIndex === undefined ||
+      scrollToIndex === null ||
+      scrollToIndex < 0
+    ) {
+      return;
+    }
+
+    rowRefs.current[page[scrollToIndex].id]?.scrollIntoView();
+  }, [enableVirtualization, page, scrollToIndex]);
```

Comment:
ok, I guess this might not work as expected, because `useEffect` will be run after `page` update, which is not what we intend. I have added `scrollToRow` prop to `LazyLoading` story and disabled virtualization. Whenever new data is loaded, you are being scrolled to the item back.

Reply:
I also thought so, but would it work in the case like...
Table is already rendered, and then user selects and element in the viewer and wants the table to scroll to that element? (technically, it is not run on init).

Reply:
It should work, if we leave the `scrollToIndex` dependency. If the user selects a row, then their scrollToIndex function should change and retrigger this, even though it would be using the original page data, right?

Reply:
Even if we remove `page` dep, it will be triggered from user, because his `scrollToIndex` will be updated with new data loaded (data is a dep of memoized `scrollToIndex` function).
So... seems its more complex now :D I know @bentleyvk was having a look today at this, but not sure he found a solution yet

Reply:
Yeah, that's a good point, it is some other variable. I guess it was a lazy story. =~)
For me, it would something like `rows.findIndex((row) => row.original.id == selectedIds[0])`, or in an earlier implementation, `selectedItem={selectedIds[0]}`

Reply:
I also have another idea, what if we pass `data` to `scrollToRow`?
```
scrollToRow={React.useCallback(
  (rows: Row<TableStoryDataType>[], data:<TableStoryDataType>) =>
    rows.findIndex((row) => row.original.id === data[12345].id),
  [],
)}
```
Then users could use `data` in their function but it won't cause rerenders -> no unnecessary scrolls.

Reply:
Now it is passing `data` to scroll function prop.

Reply:
@HaveSpacesuit @bentleyvk so what do we do with this one?
With current implementation, it looks fine to me. As there are some limitations (lazy load, paginated tables do not work), we could add maybe `@beta` tag as it is not fully implemented feature and just go with it?
@HaveSpacesuit, do you think we should just get rid of data in the callback and rely on user if he is going to include data in his calculactions, that everything is memoized and it wont cause the rerender and as a result scrolling everytime to the same row..?

Reply:
I would agree to have `@beta` tag and merge it as it is.

Reply:
I'm fine either way. If it doesn't cause issues to pass the data back, I can always just ignore it =~)

Reply:
Added `@beta` tag